agents/base/Agent.java

```diff
@@ -132,7 +132,7 @@ public void close() {
      * @param iterationScopeId An id that uniquely identifies a concept within the scope of the agent at a particular iteration
      * @return
      */
-    protected int hashcode(int iterationScopeId) {
+    protected int uniqueID(int iterationScopeId) {
```

Comment:
Whilst ID is the correct abbreviation in English, we tend to follow strict camelCase rules and stylise it as `Id` instead. The reasoning is that camel case is designed to delimit word boundaries, so `XmlIdWriter` is better than `XMLIDWriter`.

tl;dr; change this method name to `uniqueId`.